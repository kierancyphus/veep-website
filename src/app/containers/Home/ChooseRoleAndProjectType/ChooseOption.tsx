import React, { FC } from 'react'
import { Box, Typography, Button } from '@material-ui/core'
import { WithSetter } from '../resources'

export interface ChooseOptionProps extends WithSetter {
  title: string
  value: number
  toStringMap: Record<string, string>
}

export const ChooseOption: FC<ChooseOptionProps> = ({
  title,
  value,
  setValue,
  toStringMap,
}) => (
  <Box marginBottom={5}>
    <Box display="flex" justifyContent="center">
      <Typography variant="h6">{title}</Typography>
    </Box>
    <Box
      display="flex"
      justifyContent="center"
      alignContent="baseline"
      flexWrap="wrap"
    >
      {Object.entries(toStringMap).map(([type, text]) => (
        <Box margin={0.5} key={`choose-${title}-type-${text}`}>
          <Button
            onClick={() => setValue(parseInt(type))}
            variant={value === parseInt(type) ? 'contained' : 'outlined'}
            color="primary"
          >
            {text}
          </Button>
        </Box>
      ))}
    </Box>
  </Box>
)
