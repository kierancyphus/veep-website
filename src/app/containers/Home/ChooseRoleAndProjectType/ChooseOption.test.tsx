import React from 'react'
import { ChooseOption, ChooseOptionProps } from './ChooseOption'
import { render, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

describe('ChooseOption', () => {
  const props: ChooseOptionProps = {
    title: 'title',
    value: 0,
    toStringMap: {
      1: 'test',
      2: 'another test',
    },
    setValue: jest.fn(),
  }

  it('renders without errors', () => {
    const { container: firstChild } = render(<ChooseOption {...props} />, {
      wrapper: BrowserRouter,
    })

    expect(firstChild).toMatchSnapshot()
  })

  it('calls setValue with the right value on click', () => {
    const { getByText } = render(<ChooseOption {...props} />, {
      wrapper: BrowserRouter,
    })

    fireEvent.click(getByText(props.toStringMap['2']))

    expect(props.setValue).toHaveBeenCalledWith(2)
  })
})
