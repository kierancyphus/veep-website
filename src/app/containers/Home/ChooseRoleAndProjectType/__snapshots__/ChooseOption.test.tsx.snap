// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseOption renders without errors 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        title
      </h6>
    </div>
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <div
        class="MuiBox-root MuiBox-root-8"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            test
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            another test
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
